File: src/cogs/info.py
"""File holding the Cog for GitHub commands"""
import discord
from discord.ext import commands


class InfoCog(commands.Cog):
    """Github Commands Cog"""

    def __init__(self, bot: discord.Bot) -> None:
        self.bot = bot

    @commands.slash_command(
        name="info", description="Get information about yourself or another user"
    )
    async def info(self, ctx: discord.ApplicationContext, user: discord.Member) -> None:
        """Info Command"""
        user = user or ctx.author
        avatar_url = user.avatar.url if user.avatar else ""
        embed = discord.Embed()
        embed.set_author(name=user.name, icon_url=avatar_url)
        embed.set_thumbnail(url=avatar_url)
        embed.add_field(name="Username", value=user.name, inline=True)
        embed.add_field(name="Discriminator", value=user.discriminator, inline=True)
        embed.add_field(name="ID", value=str(user.id), inline=True)
        embed.add_field(
            name="Created at", value=user.created_at.strftime("%b %d, %Y"), inline=True
        )
        embed.add_field(
            name="Joined at",
            value=user.joined_at.strftime("%b %d, %Y") if user.joined_at else "N/A",
            inline=True,
        )
        embed.add_field(
            name="Status",
            value=str(user.status),
            inline=True,
        )
        embed.add_field(name="Bot", value=str(user.bot), inline=True)
        embed.add_field(name="Mention", value=user.mention, inline=True)
        embed.color = 0x0000FF  # blue color

        await ctx.respond(embed=embed, ephemeral=True)


def setup(bot: discord.Bot) -> None:
    """Adds cog to the bot"""
    bot.add_cog(InfoCog(bot))

File: src/commands/info.ts
import { EmbedBuilder, GuildMember, SlashCommandBuilder } from "discord.js";
import { Command } from "../interfaces/Command";

export const info: Command = {
  data: new SlashCommandBuilder()
    .setName("info")
    .setDescription(
      "Look up information on a user account. Defaults to current user"
    )
    .addUserOption((option) =>
      option
        .setName("user")
        .setDescription("User to look up information on")
        .setRequired(false)
    )
    .setDefaultMemberPermissions("0"),
  run: async (interaction) => {
    await interaction.deferReply({ ephemeral: true });
    const member: GuildMember =
      (interaction.options.getMember("user") as GuildMember) ??
      interaction.member;
    if (member == null) {
      console.warn(
        "There was an issue finding the member during an info command"
      );
      return;
    }
    const replyEmbed = new EmbedBuilder()
      .setThumbnail(member.displayAvatarURL())
      .addFields([
        { name: "Username", value: member.user.username, inline: true },
        {
          name: "Discriminator",
          value: member.user.discriminator,
          inline: true,
        },
        { name: "ID", value: member.id, inline: true },
        {
          name: "Created At",
          value: member.user.createdAt.toLocaleDateString(),
          inline: true,
        },
        {
          name: "Joined At",
          value: member.joinedAt?.toLocaleDateString() ?? "unknown",
          inline: true,
        },
        {
          name: "Status",
          value: member.presence?.status.toString() ?? "unknown",
          inline: true,
        },
        { name: "Is Bot", value: member.user.bot ? "Yes" : "No", inline: true },
        { name: "Mention", value: `<@${member.user.id}>`, inline: true },
      ])
      .setColor(0x0000ff);

    interaction.editReply({ embeds: [replyEmbed] });
  },
};
